### PR TITLE
Origin param

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**

### CODE REVIEW

File: MergerLogic/DataTypes/Data.cs [hunk 1]
```diff
@@ -173,7 +235,7 @@ public static Data CreateDatasource(string type, string path, int batchSize, boo
             return data;
         }
 
-        public static Data CreateDatasource(string type, string path, int batchSize, bool isBase, Extent extent, int maxZoom, int minZoom = 0, bool isOneXone = false)
+        public static Data CreateDatasource(string type, string path, int batchSize, bool isBase, Extent extent, int maxZoom, int minZoom = 0, bool isOneXone = false, TileGridOrigin? origin = null)
```
Comment (asafMasa) [inline MergerLogic/DataTypes/Data.cs:238]: why it's null? why you don't set the origin to a default on startup and prevent all the "if null" checks?

File: MergerLogic/DataTypes/Data.cs [hunk 2]
```diff
@@ -104,7 +132,22 @@ protected virtual Tile GetOneXOneTile(int z, int x, int y)
         //lazy load get tile function on first call for compatibility with null utills in contractor
         protected Tile GetTileInitilaizer(int z, int x, int y)
         {
-            this._getTile = this.isOneXOne ? this.GetOneXOneTile : this.utils.GetTile;
+            GetTileFromXYZFunction fixedGridGetTileFuntion = this.isOneXOne ? this.GetOneXOneTile : this.utils.GetTile;
+            if (this.origin == TileGridOrigin.LOWER_LEFT)
+            {
+                this._getTile = (z, x, y) =>
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerLogic/DataTypes/Data.cs:138]: static is bad for mocking

File: MergerCli/SourceParser.cs [hunk 3]
```diff
@@ -89,6 +103,19 @@ private Data ParseHttpSource(string[] args, ref int idx, int batchSize, bool isB
                 {
                     isOneXOne = true;
                 }
+                TileGridOrigin? origin = null;
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerCli/SourceParser.cs]: done

**Review by shimoncohen — CHANGES_REQUESTED** [review]

**Review by asafMasa — APPROVED** [review]

File: MergerCli/Program.cs [hunk 4]
```diff
@@ -105,9 +105,9 @@ private static void PrintHelp(string programName)
...
                                     Supported sources parameters:
                                         web sources (cant be base source):
-                                            <'xyz' / 'wmts' / 'tms'> <url template> <bbox - in format 'minX,minY,maxX,maxY'> <min zoom> <max zoom> [--1x1]
+                                            <'xyz' / 'wmts' / 'tms'> <url template> <bbox - in format 'minX,minY,maxX,maxY'> <min zoom> <max zoom> [--1x1] [--UL / --LL] 
                                         file sources:
-                                            <'fs' / 's3' / 'gpkg'> <path> [--1x1]
+                                            <'fs' / 's3' / 'gpkg'> <path> [--1x1] [--UL / --LL] 
                                         **** please note all layers must be 2X1 EPSG:4326 layers ****
                                     
                                     merge sources: {programName} <batch_size> <base source> <addiotional source> [<another source source>...]
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerCli/Program.cs:113]: fixed

File: MergerCli/Program.cs [hunk 5]
```diff
@@ -105,9 +105,9 @@ private static void PrintHelp(string programName)
 
                                     Supported sources parameters:
                                         web sources (cant be base source):
-                                            <'xyz' / 'wmts' / 'tms'> <url template> <bbox - in format 'minX,minY,maxX,maxY'> <min zoom> <max zoom> [--1x1]
+                                            <'xyz' / 'wmts' / 'tms'> <url template> <bbox - in format 'minX,minY,maxX,maxY'> <min zoom> <max zoom> [--1x1] [--UL / --LL]
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerCli/Program.cs:108]: added

File: MergerCli/SourceParser.cs [hunk 6]
```diff
@@ -66,14 +66,28 @@ private Data ParseFileSource(string[] args, ref int idx, int batchSize, bool isB
             {
                 isOneXOne = true;
             }
+            TileGridOrigin? origin = null;
+            if (optionalParams.Contains("--UL"))
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerCli/SourceParser.cs]: done

File: MergerCli/SourceParser.cs [hunk 7]
```diff
@@ -66,14 +66,28 @@ private Data ParseFileSource(string[] args, ref int idx, int batchSize, bool isB
...
+            {
+                origin = TileGridOrigin.UPPER_LEFT;
+            }
+            if (optionalParams.Contains("--UL"))
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerCli/SourceParser.cs]: fixed

File: MergerCli/SourceParser.cs [hunk 8]
```diff
@@ -66,14 +66,28 @@ private Data ParseFileSource(string[] args, ref int idx, int batchSize, bool isB
...
+            {
+                if (origin != null)
+                {
+                    throw new Exception($"layer {sourceType} {sourcePath} cant be both UL and LL");
+                }
+                origin = TileGridOrigin.UPPER_LEFT;
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerCli/SourceParser.cs]: fixed

**Pull request merged**